File: Add Code Here/C++/Linkedlist.cpp
#include <iostream>  
using namespace std;  
  
struct Node  
{  
   int data;  
   struct Node *next;  
};  
void push ( struct Node** head, int nodeData )  
{  
   struct Node* newNode1 = new Node;  
   
   newNode1 -> data = nodeData;  
   newNode1 -> next = (*head);  
   
   (*head) = newNode1;  
}  
void insertAfter ( struct Node* prevNode, int nodeData )  
{  
if ( prevNode == NULL )  
{  
   cout << "the given previous node is required,cannot be NULL";   
   return;   
      
}   
   struct Node* newNode1 =new Node;   
   newNode1 -> data = nodeData;  
   newNode1 -> next = prevNode -> next;  
    prevNode -> next = newNode1;  
}  
void append ( struct Node** head, int nodeData )  
{  
struct Node* newNode1 = new Node;  
   
struct Node *last = *head;  
newNode1 -> data = nodeData;  
newNode1 -> next = NULL;  
if ( *head == NULL )  
{  
*head = newNode1;  
return;  
}  
while ( last -> next != NULL )  
last = last -> next;  
last -> next = newNode1;  
return;  
}  
void displayList ( struct Node *node )  
{  
   while ( node != NULL )  
   {  
      cout << node -> data << "-->";  
      node = node -> next;  
   }  
   
if ( node== NULL)  
cout<<"null";   
}   
int main ()   
{  
struct Node* head = NULL;  
append ( &head, 15 );   
push ( &head, 25 );   
push ( &head, 35 );   
append ( &head, 45 );  
insertAfter ( head -> next, 55 );  
   
cout << "Final linked list: " << endl;  
displayList (head);  
   
return 0;  
}  

File: Add Code Here/GO/linked_list.go
package main

import (
    "fmt"
)

type Node struct {
    data int
    next *Node
}

type List struct {
    head *Node
}

func (l *List) add(value int) {
    newNode := &Node{data: value}

    if l.head == nil {
        l.head = newNode
        return
    }

    curr := l.head
    for curr.next != nil {
        curr = curr.next
    }

    curr.next = newNode
}

func (l *List) remove(value int) {
    if l.head == nil {
        return
    }

    if l.head.data == value {
        l.head = l.head.next
        return
    }

    curr := l.head
    for curr.next != nil && curr.next.data != value {
        curr = curr.next
    }

    if curr.next != nil {
        curr.next = curr.next.next
    }
}

func main() {
    list := &List{}
    list.add(1)
    list.add(2)
    list.add(3)
    list.add(4)

    fmt.Println("Initial List: ")
    printList(list)

    list.remove(2)
    fmt.Println("List after removing 2: ")
    printList(list)

    list.remove(4)
    fmt.Println("List after removing 4: ")
    printList(list)
}

func printList(l *List) {
    curr := l.head
    for curr != nil {
        fmt.Printf("%d ", curr.data)
        curr = curr.next
    }
    fmt.Println()
}
